eclipse-che/che-operator
# fix: do not start DWR controller if devworkspaces are not enabled

### What does this PR do?
fix: do not start DWR controller if devworkspaces are not enabled.
I'll add a bit more context soon.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/20605

### How to test this PR?
I've built the image `sleshchenko/che-operator:dwco-idling`
The test scenario with chectl operator installer is the following:
1. Deploy Che with Che Server workspace engine (`chectl server:deploy --installer=operator --platform=openshift --che-operator-image=sleshchenko/che-operator:dwco-idling`);
Expected: no CRDs are available, no DWO controller installed, DWCO is not activated (can be checked by logs)

```
2021-10-08T09:26:25.731Z	INFO	setup	DevWorkspace support disabled. Will initiate restart when CheCluster with devworkspaces enabled will appear
```

2. Modify existing CheCluster and enable DevWorkspace.
Expected: Che operator detects that change and reboot(after delay up to 1minute), install DWO, reboot, activate DWCO;

3. Modify existing CheCluster and disable DevWorkspace, restart Che Operator.
Expected: CRD is present but no CheCluster with devworkspaces exists, DWCO is not activated.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
